### PR TITLE
Dataclasses to typeddict

### PR DESCRIPTION
**Pull request by crsegerie**

### CODE REVIEW

File: kili/graphql_client.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from enum import Enum
 
 import websocket
-from six.moves import urllib
+from six.moves import urllib  # type:ignore
```
Comment (PierreLeveau) [inline kili/graphql_client.py:16]: you can remove six import, this is used for Python 2 compatibility with Python 3

File: kili/queries/asset/__init__.py [hunk 2]
```diff
@@ -80,7 +79,7 @@ def assets(
         updated_at_lte: Optional[str] = None,
         as_generator: bool = False,
         label_category_search: Optional[str] = None,
-    ) -> Union[List[dict], Generator[dict, None, None], pd.DataFrame]:
+    ) -> Union[List[dict], Generator[dict, None, None]]:
```
Comment (PierreLeveau) [inline kili/queries/asset/__init__.py:82]: why is Dataframe removed?

File: requirements_dev.txt [hunk 3]
```diff
@@ -20,3 +20,7 @@ pylint
 flake8-unused-arguments
 pyright
 linkcheckmd
+
+# notebooks tests
+nbformat
```
Comment (PierreLeveau) [inline requirements_dev.txt:25]: for local tests?

**Review by PierreLeveau — APPROVED** [review]

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
small changes and questions

**Review by PierreLeveau — APPROVED** [review]

File: kili/mutations/project/__init__.py [hunk 4]
```diff
@@ -180,7 +182,7 @@ def create_project(
         title: str,
         description: str = "",
         project_type: Optional[str] = None,
-    ):
+    ) -> Project:
```
Comment (crsegerie) [inline kili/mutations/project/__init__.py:185]: x

File: kili/queries/api_key/__init__.py [hunk 5]
```diff
@@ -40,7 +40,7 @@ def api_keys(
         first: Optional[int] = None,
         disable_tqdm: bool = False,
         as_generator: bool = False,
-    ) -> Union[List[dict], Generator[dict, None, None]]:
+    ) -> Iterable[ApiKeyType]:
```
Comment (crsegerie) [inline kili/queries/api_key/__init__.py:43]: x

File: kili/queries/asset/__init__.py [hunk 6]
```diff
@@ -80,7 +79,7 @@ def assets(
         updated_at_lte: Optional[str] = None,
         as_generator: bool = False,
         label_category_search: Optional[str] = None,
-    ) -> Union[List[dict], Generator[dict, None, None], pd.DataFrame]:
+    ) -> Iterable[Asset]:
```
Comment (crsegerie) [inline kili/queries/asset/__init__.py:82]: x

File: kili/queries/project/__init__.py [hunk 7]
```diff
@@ -50,7 +50,7 @@ def projects(
         first: Optional[int] = None,
         disable_tqdm: bool = False,
         as_generator: bool = False,
-    ) -> Union[List[dict], Generator[dict, None, None]]:
+    ) -> Iterable[Project]:
```
Comment (crsegerie) [inline kili/queries/project/__init__.py:53]: x

**Pull request merged**